librz/core/cmd/cmd_flag.c

```diff
@@ -299,115 +280,197 @@ static void spaces_list(RzSpaces *sp, RzOutputMode mode) {
 	}
 }
 
+static int cmpflag(const void *_a, const void *_b) {
+	const RzFlagItem *flag1 = _a, *flag2 = _b;
+	return (flag1->offset - flag2->offset);
+}
+
 RZ_IPI void rz_core_flag_describe(RzCore *core, ut64 addr, bool strict_offset, RzOutputMode mode) {
```

Comment:
Use `RzCmdStateOutput`.

Reply:
I didn't use in this function because it's used in other places, where the `RzCmdStateOutput` doesn't make sense, e.g. visual.

Reply:
It seems to be used only in visual.c and in cmd_flag.c and I think `RzCmdStateOutput` makes sense in both cases. You can use the `rz_cmd_state_output_*` functions to quickly setup the structures.

Reply:
At least you can pass `PJ` as an argument.